lib/asciidoctor-pdf/formatted_text/transform.rb

```diff
@@ -321,6 +326,7 @@ def build_fragment(fragment, tag_name, attrs = {})
         styles << :strikethrough
       else
         fragment.update(@theme_settings[class_name]) {|k, oval, nval| k == :styles ? oval.merge(nval) : oval } if @theme_settings.key? class_name
+        (fragment[:callback] ||= []) << TextBackgroundAndBorderRenderer
```

Comment:
We don't always want to add this callback, as it isn't always necessary. It should only be added when either a background color or both a border color and border width are set.

In other words:

```
background_color || (border_width && border_color)
```

So we'll need some key detection.

Perhaps something like:

```rb
if @theme_settings.key? class_name
  fragment.update(@theme_settings[class_name]) {|k, oval, nval| k == :styles ? oval.merge(nval) : oval }
  if fragment[:background_color] || (fragment[:border_color] && fragment[:border_width])
    ((fragment[:callback] ||= []) << TextBackgroundAndBorderRenderer).uniq!
  end 
end
```

Reply:
This should have been marked request changed. I clicked the wrong button.

Reply:
I initially did that but I realized that `TextBackgroundAndBorderRenderer` is already checking if the background and/or the border are defined.
Sure it's not really efficient since the callback will be called and will do nothing.

I will add the condition 👍

Reply:
> both a border color and border width are set.

There's no default color for the border? If I just define a border width and radius it will do nothing?

Reply:
In this context, that's correct.

There is a default border color in other places, but I don't think it makes sense to apply it here.

Reply:
> Sure it's not really efficient since the callback will be called and will do nothing.

While our renderer is inefficient, it's Prawn that isn't. When you register callbacks, it does all sorts of extra calls (as I recall).